provider/oci/environ.go

```diff
@@ -57,30 +59,116 @@ func (e *Environ) ecfg() *environConfig {
 	return e.ecfgObj
 }
 
+func (e *Environ) allInstances(tags map[string]string) ([]*ociInstance, error) {
+	compartment := e.ecfg().compartmentID()
+	request := ociCore.ListInstancesRequest{
+		CompartmentId: compartment,
+	}
+	response, err := e.Compute.ListInstances(context.Background(), request)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	ret := []*ociInstance{}
+	for _, val := range response.Items {
```

Comment:
Worth logging debug messages at the sites of ```continue``` here?

Reply:
Not really. You may have a model that spins up and tares down machines constantly, and as such, you might get a lot of results with a life cycle state of ```terminated```. There is no point in logging them. That line just filters out machines we are not interested in.